### ISSUE TITLE
validation layer on top of persistence layer to prevent saving of malformed keys

### ISSUE THREAD

**Issue by murali-shris** [atsign-foundation/at_server#821]
**Is your feature request related to a problem? Please describe.**
https://github.com/orgs/atsign-foundation/projects/8/views/57
malformed keys get synced to app and cause invalid syntax exception
**Describe the solution you'd like**
add a validation layer on top of at persistence secondary layer to prevent malformed keys from getting saved.

**Comment by murali-shris** [comment]
at_commons PR: https://github.com/atsign-foundation/at_tools/pull/214
at_server PR: https://github.com/atsign-foundation/at_server/pull/844/files
Pending work:
some of valid keys (mainly reserved keys without namespace e.g @ bob:shared_key@alice) are failing..have to fix them

**Comment by murali-shris** [comment]
```
cached:@colin:🌐_wifi_network_name:_hygiene_household**family.buzz@japanesedove9 --> KeyType.invalidKey
cached:@colin:belmont,_ca**friends.buzz@barbara --> KeyType.invalidKey
cached:@colin:can’t_believe_i_got_to_take_pictures_of_hippocampus_while_in_austin!**📸_photography.buzz@japanesedove9 --> KeyType.invalidKey
cached:@colin:how_to_say_hello_in_tamil_?**atsignco.buzz@anthony --> KeyType.invalidKey
cached:@colin:i_live_in_a_pretty_little_spot_in_belmont,_ca**company_persona_info.buzz@wonderfulpelican --> KeyType.invalidKey
cached:@colin:len_&_sue**contact_details.buzz@lentres --> KeyType.invalidKey
cached:@colin:ny_times_spelling_bee_-_can_you_guess_the_word?**contact_details.buzz@concretewasp88 --> KeyType.invalidKey
cached:@colin:today’s_call_at_10:30_am**atkrypto-atchain.buzz@andysilvey --> KeyType.invalidKey
cached:@colin:update_#1**experiment_co-founders.buzz@project_21 --> KeyType.invalidKey
cached:@colin:update_#2**experiment_co-founders.buzz@project_21 --> KeyType.invalidKey
cached:@colin:update_#3**experiment_co-founders.buzz@project_21 --> KeyType.invalidKey
cached:@colin:update_#4**experiment_co-founders.buzz@project_21 --> KeyType.invalidKey
cached:@colin:update_#5**experiment_co-founders.buzz@project_21 --> KeyType.invalidKey
cached:@colin:welcome!**friends.buzz@ barbara --> KeyType.invalidKey
```
@gkc above keys from Colin's storage are returning as invalid keys with the latest regex. Ignore the space after @ . Added the space to avoid tagging a git user
@murali-shris I've edited this to wrap as a code block; I don't believe tagging git users happens if the '@' is within a code block

**Comment by gkc** [comment]
@cconstab please review the key names above. My thoughts:
* colon definitely **should not** be allowed in the key part of the AtKey
* asterisk '*' ... maybe we can allow, but I don't like it
* comma ... maybe we can allow, but I really don't like it
* apostrophe ... maybe we can allow, but I really don't like it
* exclamation mark ... maybe we can allow, but I really don't like it
* question mark  ... maybe we can allow, but I really don't like it
* ampersand '&'  ... maybe we can allow, but I really don't like it 
* pound aka hash '#' ... maybe we can allow, but I really don't like it 

All of these exist (and will exist elsewhere) because of buzz allowing users to enter their own key names with little or no constraints

**Comment by gkc** [comment]
OK - discussed on eng standup - validation rules stand as-is. They are already being enforced client-side (since February-ish). Server-side, these keys should not be allowed to be created or updated. This should not interfere with existing malformed keys in the datastore - i.e. clients should still be able to `get` the values for those keys.

A separate discussion can be had about whether to fully delete all of the existing malformed keys in data stores.

**Comment by murali-shris** [comment]
published at_persistence_secondary_server 3.0.33 with the changes
